### PR TITLE
Testing & docs

### PR DESCRIPTION
**Pull request by fuzious**
- tests added in mocha
- docs updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,11 +193,38 @@ await matic.withdrawNFT(
   options // transaction fields
 )
 
-await matic.processExits(
-  tokenAddress, // root `token` addres
+// Transder ether to an address
```
Comment (rahuldamodar94) [inline README.md:196]: Transfer*

File: README.md [hunk 2]
```diff
@@ -365,6 +392,115 @@ await maticPOSClient.exitBatchERC1155(
   options // transaction fields, can be skipped if default options are set
 )
 
+await maticPOSClient.processExits(
```
Comment (rahuldamodar94) [inline README.md:395]: add a comment explaining what the function does

File: README.md [hunk 3]
```diff
@@ -365,6 +392,115 @@ await maticPOSClient.exitBatchERC1155(
...
+  options // transaction fields
+)
+
+// Get number of tokens that has been allowed by a user
+await maticPOSClient.getERC20Allowance(
+  userAddress, // address of the user whose details to be fetched
+  token, // Token address
+  option // transaction fields
+)
+
```
Comment (rahuldamodar94) [inline README.md:412]: add comment explaining what the function does

File: README.md [hunk 4]
```diff
@@ -365,6 +392,115 @@ await maticPOSClient.exitBatchERC1155(
...
+  rootToken, // root `token` address
+  options // transaction fields
+)
+
+// Approve Mintable ERC1155 for Deposit
+await maticPOSClient.approveMintableERC1155ForDeposit(
+  rootToken, // root `token` address
+  options // transaction fields
+)
+
```
Comment (rahuldamodar94) [inline README.md:435]: add comment

File: package.json [hunk 5]
```diff
@@ -57,8 +58,10 @@
     "web3": "^1.3.0"
   },
   "devDependencies": {
-    "@truffle/hdwallet-provider": "^1.2.0",
+    "@truffle/hdwallet-provider": "^1.0.43",
```
Comment (rahuldamodar94) [inline package.json:61]: why is this done ?

File: package.json [hunk 6]
```diff
@@ -70,13 +73,17 @@
     "eth-lib": "0.2.7",
     "ganache-cli": "^6.12.1",
     "husky": "2.4.1",
+    "i": "^0.3.6",
```
Comment (rahuldamodar94) [inline package.json:76]: ?? why this needed ?

File: package.json [hunk 7]
```diff
@@ -70,13 +73,17 @@
...
     "lint-staged": "8.2.1",
     "mocha": "^8.2.1",
     "mochapack": "^2.0.6",
+    "npm": "^7.14.0",
```
Comment (rahuldamodar94) [inline package.json:80]: ??

File: src/libs/ProofsUtil.ts [hunk 8]
```diff
@@ -240,7 +240,15 @@ export default class ProofsUtil {
         }
         receiptPromises.push(web3.eth.getTransactionReceipt(tx.hash))
       })
-      receipts = await Promise.all(receiptPromises)
+      receipts = await mapPromise(
+        receiptPromises,
+        val => {
+          return Promise.resolve(val)
```
Comment (ujjwalgupta94) [inline src/libs/ProofsUtil.ts:246]: here only return val will work, as val itself is promise.

**Review by ujjwalgupta94 — CHANGES_REQUESTED** [review]

**Review by rahuldamodar94 — APPROVED** [review]

**Review by ujjwalgupta94 — APPROVED** [review]
looks good

**Pull request merged**